bundler/lib/bundler/plugin/installer.rb

```diff
@@ -32,12 +32,27 @@ def install(names, options)
       #
       # @param [Definition] definition object
       # @return [Hash] map of names to their specs they are installed with
-      def install_definition(definition)
+      def install_definition(definition, latest = false)
         def definition.lock(*); end
-        definition.resolve_remotely!
-        specs = definition.specs
 
-        install_from_specs specs
+        if latest || definition.missing_plugins?
```

Comment:
I think you can change this to `definition.missing_specs?` and you don't need a `missing_plugins?` method. and the previous `install_from_specs` method can stay without changes because all specs should be materialized so you shouldn't need `materialized_for_all_platforms`.

Reply:
I'll look into this.

Reply:
Yes, that worked (with the exception that I had to filter out "bundler" showing up as a spec)